File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
#define UNINITIALIZED -1

class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end());
        int arrows = 0, end = UNINITIALIZED;
        for (auto& pt : points) {
            if (end == UNINITIALIZED || end < pt[0]) {
                end = pt[1];
                arrows++;
            } else {
                end = min(end, pt[1]);
            }
        }
        return arrows;
    }
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a, b) -> Integer.compare(a[0], b[0]));
        int arrows = 0, end = -1;
        for (int[] point : points) {
            if (end == -1 || end < point[0]) {
                end = point[1];
                arrows++;
            } else {
                end = Math.min(end, point[1]);
            }
        }
        return arrows;
    }
}